### PR TITLE
Feature/solc v0.6

### PR DESCRIPTION
**Pull request by krebernisak**
I've separated style changes from the upgrade, so it's easier just to focus on changes in [Update contracts from v0.4 to v0.6](https://github.com/smartcontractkit/LinkToken/pull/19/commits/56b01c4bc2691ac589dc4ddaccee3853c45f9f31) commit.

- Move current implementation to `./contracts/v0.4` folder
- Update pragma pragma solidity ^0.4.11 to ^0.6.0
- Update truffle v4 to v5
- Use @chainlink/belt to compile contracts
- Add @openzeppelin/contracts: ^3.1.0
  - Use OZ ERC20 implementation for v0.6 contracts, instead of `linkStandardToken`
- Add Prettier config and format JS files
- Improve truffle config

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,57 +6,58 @@ The total supply of the token is 1,000,000,000, and each token is divisible up t
...
-```
+The project contains [v0.4 contracts](./contracts/v0.4/) that were used for LINK Ethereum Mainnet deployment in 2017, and upgraded [v0.6 contracts](./contracts/v0.6/).
```
Comment (se3000) [inline README.md:21]:
```suggestion
The project contains [v0.4 contracts](./contracts/v0.4/) that were used for LINK Ethereum Mainnet deployment in 2017. For deployments moving forward, we use the updated [v0.6 contracts](./contracts/v0.6/) which use a more recent version of solc and the OpenZeppelin token standards. These updates include a minor ABI change around approval/allowance naming.
```

File: contracts/v0.6/LinkToken.sol [hunk 2]
```diff
@@ -0,0 +1,82 @@
+pragma solidity ^0.6.0;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
```
Comment (se3000) [inline contracts/v0.6/LinkToken.sol:3]:
```suggestion
import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
```
This is imported indirectly through LinkERC20.

File: contracts/v0.6/LinkToken.sol [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+import './token/LinkERC20.sol';
+import './ERC677Token.sol';
+
+contract LinkToken is ERC20, LinkERC20, ERC677Token {
```
Comment (se3000) [inline contracts/v0.6/LinkToken.sol:7]:
```suggestion
contract LinkToken is LinkERC20, ERC677Token {
```
This is inherited indirectly through LinkERC20.

File: contracts/v0.6/LinkToken.sol [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  uint private constant TOTAL_SUPPLY = 10**27;
+  string private constant NAME = 'ChainLink Token';
+  string private constant SYMBOL = 'LINK';
+
+  constructor() ERC20(NAME, SYMBOL)
```
Comment (se3000) [inline contracts/v0.6/LinkToken.sol:12]:
```suggestion
  constructor() LinkERC20(NAME, SYMBOL)
```

File: contracts_null/.gitignore [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# Ignore everything in this directory
```
Comment (se3000) [inline contracts_null/.gitignore:1]: What is the point of this directory? Maybe explain in this file.

**Review by se3000 — APPROVED** [review]
Looks good. Couple suggestions.

File: contracts/v0.6/LinkToken.sol (hunk 4 above)
Comment (krebernisak) [reply contracts/v0.6/LinkToken.sol:12]: This will not work as  `LinkERC20` does not have a constructor itself.

Comment (se3000) [reply contracts/v0.6/LinkToken.sol:12]: Ok, can you still call the ERC20 constructor without inheriting from it directly?

Comment (krebernisak) [reply contracts/v0.6/LinkToken.sol:12]: Yes!

**Review by se3000 — APPROVED** [review]

**Pull request merged**